ubuntu_image/classic_builder.py

```diff
@@ -77,23 +80,26 @@ def prepare_image(self):
             super().prepare_image()
 
     def populate_rootfs_contents(self):
-        src = os.path.join(self.unpackdir, 'chroot')
         dst = self.rootfs
-        for subdir in os.listdir(src):
-            shutil.move(os.path.join(src, subdir), os.path.join(dst, subdir))
-        # Remove default grub bootloader settings as we ship bootloader bits
-        # (binary blobs and grub.cfg) to a generated rootfs locally.
-        grub_folder = os.path.join(dst, 'boot', 'grub')
-        if os.path.exists(grub_folder):
-            shutil.rmtree(grub_folder, ignore_errors=True)
+        if self.args.filesystem:
+            src = self.args.filesystem
+            # 'cp -a' is faster than the python functions and makes sure all
+            # meta information is preserved.
+            os.system('cp -a ' + os.path.join(src, '*') + ' ' + dst)
```

Comment:
There is nothing wrong with using system tools for such purposes, but please use our helper 'run()' command instead. Also, we generally prefer not to use the + string concatenation syntax but instead the '{}'.format() syntax. In this case though all the arguments can be easily passed to run() as a list.